src/sync.cpp

```diff
@@ -1276,5 +1312,154 @@ bool Sync::movetolocaldebris(string* localpath)
+        return;
+    }
+
+    string localname;
+    const size_t localpathSize = localpath.size();
+
+    while (da->dnext(&localpath, &localname, client->followsymlinks))
+    {
+        auto name = localname;
+        client->fsaccess->local2name(&name);
+
+        if (localpathSize > 0)
+        {
+            localpath.append(client->fsaccess->localseparator);
+        }
+
+        localpath.append(localname);
+
+        // check if this record is to be ignored
+        if (client->app->sync_syncable(this, name.c_str(), &localpath))
+        {
+            // skip the sync's debris folder
+            if (isscannable(localpath))
+            {
+                auto localfa = std::unique_ptr<FileAccess>{client->fsaccess->newfileaccess()};
+                if ((success = localfa->fopen(&localpath, true, false)))
+                {
+                    if (localfa->type == FILENODE)
+                    {
+                        success = assignfsid(*localfa, localpath, nodes);
+                    }
+                    else if (localfa->type == FOLDERNODE)
+                    {
+                        assignfsids(success, localpath, std::move(localfa), nodes);
+                    }
+                    else
+                    {
+                        success = false;
+                        LOG_err << "Unexpected file type: " << localfa->type;
+                        assert(false);
+                    }
+                }
+                else
+                {
+                    LOG_warn << "Unable to open file: " << localpath;
+                }
+            }
+        }
+        else
+        {
+            LOG_debug << "Excluded: " << name;
+        }
+
+        localpath.resize(localpathSize);
+    }
+}
+
+bool Sync::assignfsid(FileAccess& fa, const std::string& path,
+                      const std::multiset<FileFingerprint*, FileFingerprintCmp>& nodes)
+{
+    assert(fa.type == FILENODE);
+    if (!fa.fsidvalid)
+    {
+        LOG_err << "Invalid fs id";
+        return false;
+    }
+
+    FileFingerprint ffp;
+    ffp.genfingerprint(&fa);
+    if (!ffp.isvalid)
+    {
+        LOG_err << "Invalid fingerprint";
+        return false;
+    }
+
+    const auto nodeRange = nodes.equal_range(&ffp);
+    if (std::distance(nodeRange.first, nodeRange.second) == 0)
+    {
+        return true;
+    }
+
+    // We're assigning `fa.fsid` to the node that matches the `path` passed in.
+    // If there's no match we simply assign to the first node in range.
+    auto localNode = static_cast<LocalNode*>(*nodeRange.first);
+    for (auto nodeIt = nodeRange.first; nodeIt != nodeRange.second; ++nodeIt)
+    {
+        const auto node = static_cast<LocalNode*>(*nodeIt);
+        std::string nodePath;
+        node->getlocalpath(&nodePath);
+        if (nodePath == path)
+        {
+            localNode = node;
+            break;
+        }
+    }
+    localNode->setfsid(fa.fsid);
```

Comment:
should we assign if the path doesn't match?  I think that may cause issues

Reply:
Yes, we need to assign. The matching path is just a heuristic because for the most likely case. However, it would also be correct to assign the ID to any random node in the range because we simply don't know what happened on the file system while the sync was inactive. Thoughts?